libtenzir/builtins/functions/string.cpp

```diff
@@ -70,22 +73,73 @@ class starts_or_ends_with : public virtual method_plugin {
 public:
-  starts_with() : starts_or_ends_with(true) {
+  explicit trim(std::string name) : name_{std::move(name)} {
   }
-};
 
-class ends_with final : public virtual starts_or_ends_with {
-public:
-  ends_with() : starts_or_ends_with(false) {
+  auto name() const -> std::string override {
+    return name_;
   }
+
+  auto make_function(invocation inv, session ctx) const
+    -> failure_or<function_ptr> override {
+    auto subject_expr = ast::expression{};
+    auto characters_expr = std::optional<ast::expression>{};
+    TRY(argument_parser2::method(name())
+          .add(subject_expr, "<string>")
+          .add("chars", characters_expr)
+          .parse(inv, ctx));
+    auto options = arrow::compute::TrimOptions{" \t\n\v\f\r"};
+    if (characters_expr) {
+      TRY(auto characters, const_eval(*characters_expr, ctx));
+      const auto* characters_str = caf::get_if<std::string>(&characters);
+      if (not characters_str) {
+        diagnostic::error("expected string").primary(*characters_expr).emit(ctx);
+        return failure::promise();
+      }
+      options = arrow::compute::TrimOptions{*characters_str};
+    }
+    auto fn_name = fmt::format("utf8_{}", name_);
+    return function_use::make(
+      [subject_expr = std::move(subject_expr), options = std::move(options),
+       fn_name = std::move(fn_name)](evaluator eval, session ctx) -> series {
+        auto subject = eval(subject_expr);
+        auto f = detail::overload{
+          [&](const arrow::StringArray& array) {
+            auto trimmed_array
+              = arrow::compute::CallFunction(fn_name, {array}, &options);
+            if (not trimmed_array.ok()) {
+              diagnostic::warning("{}", trimmed_array.status().ToString())
+                .primary(subject_expr)
+                .emit(ctx);
+              return series::null(string_type{}, subject.length());
+            }
+            return series{string_type{},
+                          trimmed_array.MoveValueUnsafe().make_array()};
+          },
+          [&](const auto&) {
+            diagnostic::warning("`trim` expected `string`, but got `{}`",
+                                subject.type.kind())
+              .primary(subject_expr)
+              .emit(ctx);
+            return series::null(string_type{}, subject.length());
+          },
+        };
+        return caf::visit(f, *subject.array);
+      });
+  }
+
+private:
+  std::string name_ = {};
 };
 
 } // namespace
 
 } // namespace tenzir::plugins::string
 
-TENZIR_REGISTER_PLUGIN(tenzir::plugins::string::starts_with)
-TENZIR_REGISTER_PLUGIN(tenzir::plugins::string::ends_with)
+TENZIR_REGISTER_PLUGIN(tenzir::plugins::string::starts_or_ends_with{true})
+TENZIR_REGISTER_PLUGIN(tenzir::plugins::string::starts_or_ends_with{false})
+TENZIR_REGISTER_PLUGIN(tenzir::plugins::string::trim{"trim"})
+TENZIR_REGISTER_PLUGIN(tenzir::plugins::string::trim{"ltrim"})
+TENZIR_REGISTER_PLUGIN(tenzir::plugins::string::trim{"rtrim"})
```

Comment:
Not sure about the names. JS uses `trim/trimStart/trimEnd`, while Python uses `strip/lstrip/rstrip`. Also, left and right are not technically correct (the best kind of correct), because right-to-left languages are generally trimmed from the other direction.

Reply:
I implemented my suggestion. If you have some time please take a look—it's not urgent, though.

Reply:
If the default is two trim from both ends, then I think it would make more sense if `trim(<chars>, start=false)` only trims the end. But the overall semantics are not 100% obvious.

Do you dislike `trim_start` and `trim_end`? It's more along the lines what most languages do here.

Reply:
I slightly prefer just having a single version, but tbh it's such a weak preference that I'll just implement the three separate functions.